solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/GabrielaSillis/GSLoginViewModel.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  GSLoginViewModel.swift
+//  DeliveryAppChallenge
+//
+//  Created by Gabriela Sillis on 21/07/22.
+//
+
+import Foundation
+
+protocol GSLoginViewModelDelegate: AnyObject {
+    func succeedLoginAuthentication(with session: Session)
+    func failsLoginAuthentication()
+    func startLoadingView()
+    func stopLoadingView()
+    func failsToConnectInternet()
```

Comment:
`didFailLoginAuthentication`
`didSucced`
`didFailToConnectInternet`

é uma boa quando for passado, usar o did
é parecido com o padrão do `is`no present, tipo `isEmailValid`